cli/cmd/upgrade/upgrade.go

```diff
@@ -170,11 +168,18 @@ func (c *Command) Run(args []string) int {
 
 	defer common.CloseWithError(c.BaseCommand)
 
-	if err := c.validateFlags(args); err != nil {
+	err := c.validateFlags(args)
+	if err != nil {
 		c.UI.Output(err.Error())
 		return 1
 	}
 
+	c.timeoutDuration, err = time.ParseDuration(c.flagTimeout)
+	if err != nil {
+		c.UI.Output(fmt.Sprintf("Invalid timeout: %s", err))
+		return 1
+	}
```

Comment:
I moved it out of `validateFlags` because I felt it was a side-effect that didn't meet the intention of that function. IMHO it should just validate flags and not set values on the struct.

Reply:
Ah, I see what you mean. The downside though is that we have the parsing and error handling logic twice. Maybe just documenting that it will also set the timeout in the function description or renaming it to `validateAndParseFlags` would work better?

Reply:
You know, I actually realized that we can just change the type of `flagTimeout` from `string` to `time.Duration` and then we don't need validation at all. For example:

https://github.com/hashicorp/consul-k8s/blob/c33d3bc62c396cf83749e52bdf445bcb06077b52/control-plane/subcommand/server-acl-init/command.go#L212